sebs/openwhisk/openwhisk.py

```diff
@@ -281,27 +284,53 @@ def create_function(self, code_package: Benchmark, func_name: str) -> "OpenWhisk
                         code_package.language_name,
                         code_package.language_version,
                     )
-                    subprocess.run(
-                        [
-                            *self.get_wsk_cmd(),
-                            "action",
-                            "create",
-                            func_name,
-                            "--web",
-                            "true",
-                            "--docker",
-                            docker_image,
-                            "--memory",
-                            str(code_package.benchmark_config.memory),
-                            "--timeout",
-                            str(code_package.benchmark_config.timeout * 1000),
-                            *self.storage_arguments(),
-                            code_package.code_location,
-                        ],
-                        stderr=subprocess.PIPE,
-                        stdout=subprocess.PIPE,
-                        check=True,
-                    )
+                    if code_package.language_name == 'java':
+                        subprocess.run(
+                            [
+                                *self.get_wsk_cmd(),
+                                "action",
+                                "create",
+                                func_name,
+                                "--web",
+                                "true",
+                                "--docker",
+                                docker_image,
+                                "--memory",
+                                str(code_package.benchmark_config.memory),
+                                "--timeout",
+                                str(code_package.benchmark_config.timeout * 1000),
+                                *self.storage_arguments(),
+                                code_package.code_location,
+                                "--main",
+                                "Main"
+                            ],
+                            stderr=subprocess.PIPE,
+                            stdout=subprocess.PIPE,
+                            check=True,
+                        )
+
+                    else:
+                        subprocess.run(
```

Comment:
A slightly less verbose version: just define the array of arguments for `run`, and only if language is Java, then append the two arguments of `main`.

Reply:
You're right, thanks! Fixed it, and it’s cleaner now.